File: commands/feature.py
#!/usr/bin/env python3
import os
import discord
from github import Github


async def feature(ctx, title, description):

    embed = discord.Embed(color=0xFEE9B6, title="⏳  Loading...")
    interaction = await ctx.respond(embed=embed)

    G = Github(os.getenv("GITHUB_TOKEN"))
    repo = G.get_repo("NicPWNs/MEGABOT")
    label = repo.get_label("enhancement")
    submitter = ctx.user.display_name
    description = description + f"\n\n> Submitted by `{submitter}`"
    issue = repo.create_issue(
        title=title, body=description, labels=[label], assignee="NicPWNs"
    )

    embed = discord.Embed(
        color=0xFFD882,
        title=f"💡  Created New Feature Request #{issue.number}",
        description=f"[{issue.title}](https://github.com/NicPWNs/MEGABOT/issues/{issue.number})",
    )

    await interaction.edit_original_response(embed=embed)

File: commands/feature.ts
import {
  DiscordInteraction,
  InteractionResponse,
  Colors,
} from "../types/discord";
import { embedResponse, errorResponse } from "../utils/discord";
import { getOptionValue, getUser } from "../handlers/commands";
import { Octokit } from "@octokit/rest";

export async function handleFeature(
  interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  const title = getOptionValue<string>(interaction, "title")!;
  const description = getOptionValue<string>(interaction, "description")!;
  const user = getUser(interaction);

  const githubToken = process.env.GITHUB_TOKEN;
  if (!githubToken) {
    return errorResponse("GitHub integration not configured.");
  }

  try {
    const octokit = new Octokit({ auth: githubToken });

    const body = `${description}\n\n> Submitted by \`${user.username}\``;

    const { data: issue } = await octokit.issues.create({
      owner: "NicPWNs",
      repo: "MEGABOT",
      title,
      body,
      labels: ["enhancement"],
      assignees: ["NicPWNs"],
    });

    return embedResponse({
      title: `✨ Created Feature Request #${issue.number}`,
      description: `[${issue.title}](${issue.html_url})`,
      color: Colors.BLUE,
    });
  } catch (error) {
    console.error("GitHub API error:", error);
    return errorResponse(
      "Failed to create feature request. Please try again later.",
    );
  }
}
